tezos_client/src/client.rs

```diff
@@ -0,0 +1,107 @@
+use failure::Fail;
+
+use networking::p2p::binary_message::Hexable;
+use networking::p2p::encoding::prelude::*;
+use tezos_encoding::hash::{BlockHash, ChainId};
+use tezos_interop::ffi;
+use tezos_interop::ffi::OcamlStorageInitInfo;
+
+#[derive(Debug, Fail)]
+pub enum ApplyBlockError {
+    #[fail(display = "incomplete operations")]
+    IncompleteOperations,
+    #[fail(display = "failed to apply block")]
+    FailedToApplyBlock,
+    #[fail(display = "Unknown predecessor")]
+    UnknownPredecessor,
+}
+
+pub struct TezosStorageInitInfo {
+    pub chain_id: ChainId,
+    pub genesis_block_header_hash: BlockHash,
+    pub current_block_header_hash: BlockHash
+}
+
+impl TezosStorageInitInfo {
+    fn new(storage_init_info: OcamlStorageInitInfo) -> Self {
+        TezosStorageInitInfo {
+            chain_id: hex::decode(storage_init_info.chain_id).unwrap(),
+            genesis_block_header_hash: hex::decode(storage_init_info.genesis_block_header_hash).unwrap(),
+            current_block_header_hash: hex::decode(storage_init_info.current_block_header_hash).unwrap(),
+        }
+    }
+}
+
+//        TODO: priznak do storage, ze mame block zapisany v tezos-ocaml-storage
+//        TODO: spravit benches pre bootstrap test a zoptimalizovat ocaml s logovanim a bez logovania
+//        TODO: jira - od coho zavisi konfiguracia alphanet vs mainnet vs zeronet: storage a genesis?
+//                   - Return supported network protocol version
+//        TODO: jira - Tezos_validation.Block_validation.apply
+//        TODO: jira - podpora pre test_chain
+//        TODO: jira - overit proof_of_work_stamp pri bootstrape
+//        TODO: jira - pre generovanie identity
+
+/// Initializes storage for Tezos ocaml storage in chosen directory
+pub fn init_storage(storage_data_dir: String) -> TezosStorageInitInfo {
```

Comment:
Create new error type and return that error in a result `Result<(), NewErrorType>`. Or repurpose `ApplyBlockError` to also cover all possible ffi errors.